defuse/src/accounts.rs

```
@@ -31,3 +32,23 @@ pub trait AccountManager {
     /// NOTE: MUST attach 1 yⓃ for security purposes.
     fn invalidate_nonces(&mut self, nonces: Vec<AsBase64<Nonce>>);
 }
+
+#[ext_contract(ext_force_account_locker)]
+pub trait AccountForceLocker: AccessControllable {
+    /// Returns whether the given`account_id` is locked
+    fn is_account_locked(&self, account_id: &AccountId) -> bool;
+
+    /// Locks given `account_id` from modifying its own state, including
+    /// token balances.
+    /// Returns `false` if the account was already in locked state.
+    ///
+    /// Attached deposit of 1yN is required for security purposes.
+    ///
+    /// NOTE: this still allows for force withdrawals/transfers
+    fn force_lock_account(&mut self, account_id: AccountId) -> bool;
```

Instead of returning a bool here (and in unlock), we should probably just panic with a message "already locked".